lib/ood_core/job/adapters/kubernetes/helper.rb

```diff
@@ -263,25 +263,29 @@ def submission_time(json_data)
   def pod_status_from_json(json_data)
     phase = json_data.dig(:status, :phase)
     conditions = json_data.dig(:status, :conditions)
+    container_statuses = json_data.dig(:status, :containerStatuses)
     unschedulable = conditions.to_a.any? { |c| c.dig(:reason) == "Unschedulable" }
-    state = case phase
-            when "Running"
-              "running"
-            when "Pending"
-              if unschedulable
-                "queued_held"
-              else
-                "queued"
-              end
-            when "Failed"
-              "suspended"
-            when "Succeeded"
-              "completed"
-            when "Unknown"
-              "undetermined"
+    ready = container_statuses.to_a.any? { |s| s.dig(:ready) == true }
```

Comment:
Shouldn't this be `!container_statuses.empty? && container_statuses.to_a.all? { |s| s.dig(:ready) == true }` to account for other containers (that we don't yet support but, maybe will and we'll lose this).

I'm not in love with an if block and a case statemement, but I think the only other alternative is a return statement with is also not great.

Reply:
So the `.all?` would only be useful in context of multiple containers, not init containers because the key for init containers is `initContainerStatuses`.  But having multiple containers is something needing to be supported so I can update that.

Reply:
Made this change, also made it so the info function only returns string, so could return early when ready. Then the place needing a status object is what sets the status.

Reply:
There were also many places where `pod_status_from_json` was getting used by helper functions and expecting a lowercase string for comparison so to ensure that actually works and is clearly going to work, moved the `OodCore::Job::Status.new` call to the place where that object type is actually needed so `pod_status_from_json` just returns a status string.